ddtrace/tracer/spancontext_test.go

```diff
@@ -542,6 +543,91 @@ func TestSpanPeerService(t *testing.T) {
 	}
 }
 
+func TestSpanDDBaseService(t *testing.T) {
+	run := func(t *testing.T, tracerOpts []StartOption, spanOpts []StartSpanOption) []*span {
+		prevSvc := globalconfig.ServiceName()
+		t.Cleanup(func() { globalconfig.SetServiceName(prevSvc) })
+
+		tracer, transport, flush, stop := startTestTracer(t, tracerOpts...)
+		t.Cleanup(stop)
+
+		p := tracer.StartSpan("parent-span", spanOpts...)
+		childSpanOpts := append([]StartSpanOption{ChildOf(p.Context())}, spanOpts...)
+		s := tracer.StartSpan("child-span", childSpanOpts...)
+		s.Finish()
+		p.Finish()
+
+		flush(1)
+		traces := transport.Traces()
+		require.Len(t, traces, 1)
+		require.Len(t, traces[0], 2)
+
+		return traces[0]
+	}
+	t.Run("span-service-not-equal-global-service", func(t *testing.T) {
+		tracerOpts := []StartOption{
+			WithService("global-service"),
+		}
+		spanOpts := []StartSpanOption{
+			ServiceName("span-service"),
+		}
+		spans := run(t, tracerOpts, spanOpts)
+		for _, s := range spans {
+			assert.Equal(t, "span-service", s.Service)
+			assert.Equal(t, "global-service", s.Meta["_dd.base_service"])
+		}
+	})
+	t.Run("span-service-equal-global-service", func(t *testing.T) {
+		tracerOpts := []StartOption{
+			WithService("global-service"),
+		}
+		spanOpts := []StartSpanOption{
+			ServiceName("global-service"),
+		}
+		spans := run(t, tracerOpts, spanOpts)
+		for _, s := range spans {
+			assert.Equal(t, "global-service", s.Service)
+			assert.NotContains(t, s.Meta, "_dd.base_service")
+		}
+	})
+	t.Run("span-service-equal-different-case", func(t *testing.T) {
+		tracerOpts := []StartOption{
+			WithService("global-service"),
+		}
+		spanOpts := []StartSpanOption{
+			ServiceName("GLOBAL-service"),
+		}
+		spans := run(t, tracerOpts, spanOpts)
+		for _, s := range spans {
+			assert.Equal(t, "GLOBAL-service", s.Service)
+			assert.NotContains(t, s.Meta, "_dd.base_service")
+		}
+	})
+	t.Run("global-service-not-set", func(t *testing.T) {
+		spanOpts := []StartSpanOption{
+			ServiceName("span-service"),
+		}
+		spans := run(t, nil, spanOpts)
+		for _, s := range spans {
+			assert.Equal(t, "span-service", s.Service)
+			assert.NotEmpty(t, s.Meta["_dd.base_service"])
```

Comment:
What's the expected value here? Can we use `assert.Equal`?

Reply:
This is the fallback default service name value, that we obtain calling `filepath.Base(os.Args[0])`. I didn't want to `assert.Equal` to any concrete value because it might change depending on how you run the tests.

I guess an option would be to `assert.Equal(t, filepath.Base(os.Args[0]), s.Meta["_dd.base_service"])`, but then the test would fail in case we decide to change this logic some day. What do you think?

Reply:
Thanks for explaining, I don't think we should leak the implementation details of the defaulting to this unit test. So you can keep the code as is and just add a comment to say why we are not asserting with `Equal`.